### DISCUSSION TITLE
fish scripts should only run in interactive shells

### DISCUSSION THREAD

**Discussion by injust** [aws/codewhisperer-command-line-discussions#226] [Feature request]
### Checks

- [X] I could not find an existing setting that solved my issue
- [X] I searched [the discussions in this repo](https://github.com/aws/q-command-line-discussions/discussions?q=) and there are no duplicates of my feature request

### Description
This should be easy, just add this to `00_fig_pre.fish` and `99_fig_post.fish`:
```fish
if status is-interactive
    ...
end
```

**Comment by grant0417** [comment]
What part of the integrations are running in non interactive script? If you could give a reproduction of what is causing you issues I can help resolve/make your modification.

**Reply by injust** [reply]
The integrations aren't running when it's non-interactive (I saw the check in `~/.local/bin/q init fish pre`).

But it's better to be explicit and fail early -- why not immediately exit so you don't run `~/.local/bin/q` at all? Hence the suggestion to add the `status is-interactive` check in `00_fig_pre.fish` and `99_fig_post.fish`.
